src/oc/storage/api/entries.clj

```diff
@@ -361,6 +379,69 @@
   :handle-unprocessable-entity (fn [ctx]
     (api-common/unprocessable-entity-response (:reason ctx))))
 
+;; A resource for reverting to a specific revision number.
+(defresource revert-version [conn org-slug board-slug entry-uuid]
+  (api-common/open-company-authenticated-resource config/passphrase) ; verify validity and presence of required JWToken
+
+  :allowed-methods [:options :post]
+
+  ;; Authorization
+  :allowed? (by-method {
+    :options true
+    :post (fn [ctx] (access/allow-authors conn org-slug (:user ctx)))})
+
+  ;; Media type client accepts
+  :available-media-types (by-method {
+                            :post [mt/entry-media-type]})
+  :handle-not-acceptable (by-method {
+                            :post (api-common/only-accept 406 mt/entry-media-type)})
+
+  ;; Possibly no data to handle
+  :malformed? (by-method {
+    :options false
+    :post (fn [ctx] (api-common/malformed-json? ctx true))}) ; allow nil
+  :processable? (by-method {
+    :options true
+    :post (fn [ctx] (let [entry-props (:data ctx)]
+                      (valid-entry-revert? entry-props)))})
+  :new? false
+  :respond-with-entity? true
+
+  ;; Existentialism
+  :can-post-to-missing? false
+  :exists? (fn [ctx]
+             (if-let* [_slugs? (and (slugify/valid-slug? org-slug)
+                                    (slugify/valid-slug? board-slug))
+                       org (org-res/get-org conn org-slug)
+                       org-uuid (:uuid org)
+                       entry (entry-res/get-entry conn entry-uuid)
+                       existing-version (if (zero? (:revision-id (:data ctx)))
```

Comment:
I am not sure this works... If the user is asking to revert to revision 0 (ie the first revision) we return the current revision version.
But what if the current revision version is 5?

Reply:
The first version (version 0) only exists in the `entries` table.  If a user asks to revert to zero, we use the existing entry data. There is a test in the revert function that deletes entry versions if the revision id is zero.  I could just use the `:uuid` in this case instead of all the entry info.